tests/31sync/05presence.pl

```diff
@@ -0,0 +1,81 @@
+use List::Util qw( first );
+
+test "User sees their own presence in a sync",
+    requires => [qw( first_api_client can_sync )],
+
+    check => sub {
+        my ( $http ) = @_;
+        my ( $user, $filter_id );
+        matrix_register_user_with_filter( $http, {} )->then( sub {
+            ( $user, $filter_id ) = @_;
+            matrix_sync( $user, filter => $filter_id )
+        })->then( sub {
+            my ( $body ) = @_;
+            my $events = $body->{presence}{events};
+            my $presence = first { $_->{type} eq "m.presence" } @$events;
+            defined $presence or die "Expected to see our own presence";
+            $presence->{sender} eq $user->user_id or die "Unexpected sender";
+            $presence->{content}{presence} eq "online"
+                or die "Expected to be online";
+            Future->done(1)
+        })
+    };
+
+test "User is offline if they set_presence=offline in their sync",
+    requires => [qw( first_api_client can_sync )],
+
+    check => sub {
+        my ( $http ) = @_;
+        my ( $user, $filter_id );
+        matrix_register_user_with_filter( $http, {} )->then( sub {
+            ( $user, $filter_id ) = @_;
+            matrix_sync( $user, filter => $filter_id, set_presence => "offline")
+        })->then( sub {
+            my ( $body ) = @_;
+            my $events = $body->{presence}{events};
+            my $presence = first { $_->{type} eq "m.presence" } @$events;
+            defined $presence or die "Expected to see our own presence";
+            $presence->{sender} eq $user->user_id or die "Unexpected sender";
+            $presence->{content}{presence} eq "offline"
+                or die "Expected to be offline";
+            Future->done(1)
+        })
+    };
+
+test "User sees updates to presence from other users in the incremental sync.",
+    requires => [qw( first_api_client can_sync )],
+
+    check => sub {
+        my ( $http ) = @_;
+        my ( $user_a, $user_b, $filter_id_a, $filter_id_b, $next_a );
+        Future->needs_all(
+            matrix_register_user_with_filter( $http, {} ),
+            matrix_register_user_with_filter( $http, {} ),
+        )->then( sub {
+            ( $user_a, $filter_id_a, $user_b, $filter_id_b ) = @_;
+            # We can't use matrix_create_and_join since that polls the event
+            # stream to check that the user has joined.
```

Comment:
This issue has come up a few times now. I wonder if that should be an option to `matrix_create_and_join_room()` to not do the event stream